pydis_site/apps/api/tests/test_reminders.py

```diff
@@ -0,0 +1,139 @@
+from datetime import datetime
+
+from django_hosts.resolvers import reverse
+
+from .base import APISubdomainTestCase
+from ..models import Reminder, User
+
+
+class UnauthedReminderAPITests(APISubdomainTestCase):
+    def setUp(self):
+        super().setUp()
+        self.client.force_authenticate(user=None)
+
+    def test_list_returns_401(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 401)
+
+    def test_create_returns_401(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.post(url, data={'not': 'important'})
+
+        self.assertEqual(response.status_code, 401)
+
+    def test_delete_returns_401(self):
+        url = reverse('bot:reminder-detail', args=('1234',), host='api')
+        response = self.client.delete(url)
+
+        self.assertEqual(response.status_code, 401)
+
+
+class EmptyDatabaseReminderAPITests(APISubdomainTestCase):
+    def test_list_all_returns_empty_list(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), [])
+
+    def test_delete_returns_404(self):
+        url = reverse('bot:reminder-detail', args=('1234',), host='api')
+        response = self.client.delete(url)
+
+        self.assertEqual(response.status_code, 404)
+
+
+class ReminderCreationTests(APISubdomainTestCase):
+    def setUp(self):
+        super().setUp()
+        self.author = User.objects.create(
+            id=1234,
+            name='Mermaid Man',
+            discriminator=1234,
+            avatar_hash=None,
+        )
+        self.data = {
+            'author': self.author.id,
+            'content': 'Remember to...wait what was it again?',
+            'expiration': datetime.utcnow().isoformat(),
+            'jump_url': "https://www.google.com",
+            'channel_id': 123,
+        }
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.post(url, data=self.data)
+        self.assertEqual(response.status_code, 201)
```

Comment:
I thought that was odd too. I used a couple existing files created by respected contributors as models for mine, and this "assert within setUp()" appeared in a couple places. Would it be appropriate to move this into the `test_reminder_in_full_list` test? Or is it better to test creation using a `post` against the list endpoint in one test, and use another test to check that a `get` for the list endpoint returns expected reminder instances?

Reply:
Put it in a new test function. As per [this](https://github.com/python-discord/site/pull/322#discussion_r373799373) comment, put that assertion in this new function too and `test_reminder_in_full_list` can be removed completely.

Reply:
Actually, it may be moved to another test case since retrieval tests are needed. It just doesn't belong with the creation tests at least.

> Needs a test case for the list endpoint to test retrieval of existing objects and also test searching and filtering.